content/v2.0/application/settings.md

```diff
@@ -0,0 +1,266 @@
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :analytics_enabled, constructor: Types::Params::Bool.optional
+    setting :max_cart_items, constructor: Types::Params::Integer.optional
+  end
+end
+```
+
+## Default values
+
+Settings can also specify defaults to be used in the absence of the relevant environment variable.
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    setting :redis_url, default: "redis://localhost:6379", constructor: Types::String
+    setting :analytics_enabled, default: false, constructor: Types::Params::Bool
+    setting :max_cart_items, default: 100, constructor: Types::Params::Integer
+  end
+end
+```
+
+## Constraints
+
+To enforce additional contraints on settings, you can use a [dry-types](https://dry-rb.org/gems/dry-types) constraint in your constructor.
+
+Here, the value of the `session_secret` must be at least 32 characters, while the value of the `from_email` setting must satisfy the EMAIL_FORMAT regular expression:
+
+```ruby
+# config/settings.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class Settings < Hanami::Settings
+    EMAIL_FORMAT = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z]+)*\.[a-z]+\z/i
+
+    setting :session_secret, constructor: Types::String.constrained(min_size: 32)
+    setting :from_email, constructor: Types::String.constrained(format: EMAIL_FORMAT)
+  end
+end
+```
+
+## Using settings within your application
+
+Hanami makes settings available within your application through a built-in settings provider.
+
+```ruby
+Hanami.app["settings"]
+```
+
+### Accessing settings within components
+
+To access settings within one of your components, use Hanami's Deps mixin:
+
+```ruby
+# app/analytics/send_event.rb
+
+module Bookshelf
+  module Analytics
+    class SendEvent
+      include Deps["settings"]
+
+      def call(event_type, payload)
+        return unless settings.analytics_enabled
+
+        # ...send event to analytics service here ...
+      end
+    end
+  end
+end
+```
+
+For more information on components and the Deps mixin, see the [architecture guide](/v2.0/architecture/containers/).
+
+### Accessing settings within providers
+
+When registering a provider, settings can be access via the `target` method, which exposes the application's container:
+
+```ruby
+# config/providers/redis.rb
+
+# frozen_string_literal: true
+
+Hanami.app.register_provider :redis do
+  start do
+    require "redis"
+
+    redis = Redis.new(url: target["settings"].redis_url)
+
+    register "redis", redis
+  end
+end
+```
+
+See the guides for [providers](/v2.0/architecture/providers/) and [containers](/v2.0/architecture/containers/) for more information.
+
+### Accessing settings within your app class
+
+In some cases, you may want to use a setting to inform an application configuration inside your `App` class. Within the `App` class, settings are exposed at the class level for you, via a `settings` method:
```

Comment:
Perhaps after the code example we could have a short note saying "Because settings can be accessed at this early point in the app's boot, it's important that the settings file is self-contained, with no dependencies on other code in your app." (Happy for you to finesse the wording)

Reply:
Thanks - added a note to this effect.